tests/test_transformations/test_normalize_features.py

```diff
@@ -0,0 +1,96 @@
+import unittest
+import pandas as pd
+from anndata import AnnData
+from spac.transformations import normalize_features
+
+
+class TestNormalizeFeatures(unittest.TestCase):
+
+    def setUp(self):
+        # Create a sample AnnData object with a test dataframe
+        self.data = {
+            'feature1': [0.1, 0.2, 0.3, 0.4, 0.5],
+            'feature2': [0.6, 0.7, 0.8, 0.9, 1.0]
+        }
+        self.adata = AnnData(pd.DataFrame(self.data))
+
+    def test_invalid_high_quantile(self):
+        # Test with an invalid high_quantile (not numeric)
+        with self.assertRaises(TypeError):
+            normalize_features(self.adata, high_quantile='2')
+
+    def test_invalid_low_quantile(self):
+        # Test with an invalid low_quantile (not numeric)
+        with self.assertRaises(TypeError):
+            normalize_features(self.adata, low_quantile='2')
+
+    def test_high_quantile_out_of_range(self):
+        # Test with high_quantile out of range (should be within (0, 1])
+        with self.assertRaises(ValueError):
+            normalize_features(self.adata, high_quantile=1.5)
+
+    def test_low_quantile_out_of_range(self):
+        # Test with low_quantile out of range (should be within [0, 1))
+        with self.assertRaises(ValueError):
+            normalize_features(self.adata, low_quantile=-0.5)
+
+    def test_low_quantile_greater_than_high_quantile(self):
+        # Test with low_quantile greater than high_quantile
+        with self.assertRaises(ValueError):
+            normalize_features(
+                self.adata,
+                low_quantile=0.8,
+                high_quantile=0.7
+            )
+
+    def test_correct_output(self):

```

Comment:
I suggest to use a scenario that is easy to reason about, e.g, inputs are from 0 to 1, or 100 to 200. This way a developer can know the "expected output" right away without calculating it.